### PR TITLE
GEODE-7957: query results toData will write to correct output stream 

### PR DESCRIPTION
**Pull request by jhuynh1**
* Previously, when a query is executed in a function, the toData on specific results
    set types would write directly to the wrong output stream, causing deserialization issues

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/cache/query/internal/CumulativeNonDistinctResults.java [hunk 1]
```diff
@@ -177,6 +177,7 @@ public CumulativeNonDistinctResultsCollection(Collection<? extends Collection<E>
       this.results = results;
       this.limit = limit;
       this.collectionsMetdata = collectionsMetadata;
+      System.out.println("JASON");
```
Comment (jujoramos) [inline geode-core/src/main/java/org/apache/geode/cache/query/internal/CumulativeNonDistinctResults.java:180]: You might want to remove this one :-)

File: geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt [hunk 2]
```diff
@@ -282,74 +282,6 @@ toData,12
...
-toData,25
-
-org/apache/geode/distributed/internal/membership/gms/messages/SuspectMembersMessage,2
-fromData,63
-toData,92
-
-org/apache/geode/distributed/internal/membership/gms/messages/ViewAckMessage,2
-fromData,40
-toData,37
-
```
Comment (jujoramos) [inline geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt:352]: Did you intentionally remove all the above `sanctionedDataSerializables.txt `?, they don't seem related to the issu e fixed by this PR.

File: geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt [hunk 3]
```diff
@@ -358,14 +290,6 @@ org/apache/geode/distributed/internal/streaming/StreamingOperation$StreamingRepl
...
 toData,86
 
-org/apache/geode/distributed/internal/tcpserver/InfoResponse,2
-fromData,9
-toData,9
-
-org/apache/geode/distributed/internal/tcpserver/VersionResponse,2
-fromData,11
-toData,11
-
```
Comment (jujoramos) [inline geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt:368]: Same as with the others: should these be removed?.

File: geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt [hunk 4]
```diff
@@ -578,9 +502,6 @@ org/apache/geode/internal/admin/remote/LicenseInfoResponse,2
 fromData,18
 toData,15
 
-org/apache/geode/internal/admin/remote/MissingPersistentIDsRequest,1
-fromData,7
-
```
Comment (jujoramos) [inline geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/sanctionedDataSerializables.txt:583]: Same as with the others: should this be removed?.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+  @Rule
+  public ClientCacheRule clientCacheRule = new ClientCacheRule();
+
+  private VM server1, server2, client;
+  private String regionName = "region";
```
Comment (jujoramos) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java:65]: This one can be declared as `static final REGION_NAME` as it's a constant.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    server2 = getVM(1);
+    client = getVM(2);
+  }
+
+  @Test
+  public void cumulativeResultsToDataShouldWriteToTheCorrectStreamNotCauseCorruption() {
+    int numPuts = 10;
+    int port = server1.invoke(() -> {
+      return createServerAndRegion();
+    });
```
Comment (jujoramos) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java:79]: Can be replaced with `int port = server1.invoke(this::createServerAndRegion);`.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    server2.invoke(() -> {
+      createServerAndRegion();
+    });
```
Comment (jujoramos) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java:82]: Can be replaced with `server2.invoke((SerializableRunnableIF) this::createServerAndRegion);`.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.dunit;
+
+import static org.apache.geode.test.dunit.VM.getVM;
+import static org.junit.Assert.assertEquals;
```
Comment (jujoramos) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java:18]: We are replacing usages of JUnit `asserts` for `assert4j`, we can change this before merging the pull request to make sure we don't need to revisit this test in the future.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+    public TestObject(int id) {
+      this.id = id;
+      this.name = "Name:" + id;
+    }
+  }
+
+  public static class QueryWithoutTurningIntoListFunction extends FunctionAdapter {
+
+    private String regionName;
```
Comment (jujoramos) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/MultiServerPartitionedRegionQueryDUnitTest.java:157]: This field is never used.

**Comment by jhuynh1** [comment]
> LGTM, just some nitpicks, feel free to ignore them.
> One thing I've noted is that both test methods (`cumulativeResultsToDataShouldWriteToTheCorrectStreamNotCauseCorruption` and `nwayMergeResultsToDataShouldWriteToTheCorrectStreamAndNotCauseCorruption`) _**do exactly the same thing**_, am I missing something?, or can we just delete one of them?.
Good catch, I must have accidently modified one to match the other by mistake.  The query should be modified to generate an N-Way result and the other the Cumulative type.
